extensions/spring-cloud.md

```diff
@@ -1,51 +1,89 @@
+The sole full implementation provided of the `CapabilityDiscoveryMode`, is the `RestCapabilityDiscoveryMode`, using a `RestTemplate` and the `ServiceInstance` URI to invoke a configurable endpoint.
+This endpoint leads to the `MemberCapabilitiesController` which in turn exposes the `MemberCapabilities` on the `RestCapabilityDiscoveryMode` of that instance.
 
-The `SpringCloudCommandRouter` has to be created by providing the following:
+There are decorators present for the `CapabilityDiscoveryMode`, providing two additional features:
 
-* A "discovery client" of type `DiscoveryClient` - This can be provided by annotating your Spring Boot application with `@EnableDiscoveryClient`, which will look for a Spring Cloud implementation on your classpath.
-* A "routing strategy" of type `RoutingStrategy` - The `axon-messaging` module currently provides several implementations, but a function call can suffice as well.
+1. `IgnoreListingDiscoveryMode` - a `CapabilityDiscoveryMode` decorator which on failure of retrieving the `MemberCapabilities` will place the given `ServiceInstance` on a list to be ignored for future validation. It thus effectively removes discoverable `ServiceInstance`s from the set.
+2. `AcceptAllCommandsDiscoveryMode` - a `CapabilityDiscoveryMode` decorator which regardless of what _this_ instance can handle as commands, state it can handle anything. This decorator comes in handy if the nodes in the system are homogeneous (aka, everybody can handle the same set of commands). 
 
-  If you want to route the commands based on the 'aggregate identifier' for example, you would use the `AnnotationRoutingStrategy` and annotate the field on the payload that identifies the aggregate with `@TargetAggregateIdentifier`.
+The `Registration`, `DiscoveryClient` and `CapabilityDiscoveryMode` are arguably the heart of the `SpringCloudCommandRouter`.
+There are however a couple of additional things you can configure for this router, which are the following:
 
-* A "local service instance" of type `Registration` - If you're Spring Boot application is annotated with the aforementioned `@EnableDiscoveryClient`, it will automatically create a `Registration` bean referencing the instance itself.
+* `RoutingStrategy` - The component in charge of deciding which of the nodes receives the commands consistently. By default a `AnnotationRoutingStrategy` is used (see [Distributing the Command Bus](../axon-framework/axon-framework-commands/implementations.md#distributedcommandbus) for more).
+* A `ServiceInstance` filter - This `Predicate` is used to filter out `ServiceInstance`s retrieved through the `DiscoveryClient`. For example allows the removal of instances which are known to not handle any command messages. This might be useful if you've got several services within the Spring Cloud Discovery Service set up which you do not want to take into account for command handling, ever.
```

Comment:
```suggestion
* A `ServiceInstance` filter - This `Predicate` is used to filter out `ServiceInstance`s retrieved through the `DiscoveryClient`. For example, it allows the removal of instances which are known to not handle any command messages. This might be useful if you have several services within the Spring Cloud Discovery Service set up, which you do not ever want to take into account for command handling.
```